pkg/geo/geomfn/add_measure.go

```diff
@@ -0,0 +1,106 @@
+// Copyright 2021 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package geomfn
+
+import (
+	"github.com/cockroachdb/cockroach/pkg/geo"
+	"github.com/cockroachdb/errors"
+	"github.com/twpayne/go-geom"
+)
+
+func AddMeasureToLineString(
+	geometry geo.Geometry, start float64, end float64,
+) (geo.Geometry, error) {
+	t, err := geometry.AsGeomT()
+	if err != nil {
+		return geometry, err
+	}
+
+	switch t := t.(type) {
+	case *geom.LineString: //, *geom.MultiLineString: TODO have a function that calls this multiple times
+		// Add an M dimension to the linestring if it doesn't have one already.
+		geometryWithM, err := ForceLayout(geometry, augmentLayoutWithM(t.Layout()))
```

Comment:
i would look at doing this without `ForceLayout`, i.e. reconstructing a flat coords and calling appending an `M` coordiante onto the layout before calling `geom.NewLineStringFlat`.